### PR TITLE
Support ServerlessConnection in Embedding

### PR DESCRIPTION
**Pull request by dorisjoy**
Support ServerlessConnection in Embedding

### CODE REVIEW

File: src/promptflow-tools/promptflow/tools/embedding.py [hunk 1]
```diff
@@ -10,6 +10,14 @@
 from promptflow.connections import AzureOpenAIConnection, OpenAIConnection
 
 
+def is_serverless_connection(connection):
```
Comment (chjinche) [inline src/promptflow-tools/promptflow/tools/embedding.py:13]: Use this compatible process? Then you can keep using `isinstance`.
try:
    from promptflow.connections import ServerlessConnection
except ImportError:
    # If unable to import ServerlessConnection, define a placeholder class to allow isinstance checks to pass.
    # ServerlessConnection was introduced in pf version 1.6.0.
    class ServerlessConnection:
        pass

File: src/promptflow-tools/promptflow/tools/embedding.py [hunk 2]
```diff
@@ -33,6 +45,13 @@ def embedding(connection: Union[AzureOpenAIConnection, OpenAIConnection], input:
             input=input,
             model=model
         ).data[0].embedding
+    elif is_serverless_connection(connection):
```
Comment (chjinche) [inline src/promptflow-tools/promptflow/tools/embedding.py:48]: do we plan to add ut in this pr?

Comment (melionel) [reply src/promptflow-tools/promptflow/tools/embedding.py:48]: Yes, we should. I want to add the ut after I tested in local. However I didn't get an embedding deployment for test now.

File: src/promptflow-tools/promptflow/tools/embedding.py [hunk 3]
```diff
@@ -18,8 +25,12 @@ class EmbeddingModel(str, Enum):
 
 @tool
 @handle_openai_error()
-def embedding(connection: Union[AzureOpenAIConnection, OpenAIConnection], input: str, deployment_name: str = "",
-              model: EmbeddingModel = EmbeddingModel.TEXT_EMBEDDING_ADA_002):
+def embedding(
+    connection: Union[AzureOpenAIConnection, OpenAIConnection],
```
Comment (DaweiCai) [inline src/promptflow-tools/promptflow/tools/embedding.py:29]: Union[AzureOpenAIConnection, OpenAIConnection, ServerlessConnection] ？

Comment (melionel) [reply src/promptflow-tools/promptflow/tools/embedding.py:29]: good catch

**Review by DaweiCai — APPROVED** [review]

**Review by chjinche — APPROVED** [review]

**Pull request merged**